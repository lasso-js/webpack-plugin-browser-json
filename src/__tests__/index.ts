import fs from "mz/fs";
import path from "path";
import { BrowserJSONPlugin } from "..";
import { compile } from "./utils/compile";

const FIXTURES = path.join(__dirname, "fixtures");

for (const fixture of fs.readdirSync(FIXTURES)) {
  test(fixture, async () => {
    const fixtureEntry = path.join(FIXTURES, fixture, "index.js");
    const flagsFile = path.join(FIXTURES, fixture, "flags.json");
    const flags = (await fs.exists(flagsFile))
      ? JSON.parse(await fs.readFile(flagsFile, "utf-8"))
      : {};
    await expect(
      compile(fixtureEntry, new BrowserJSONPlugin({ flags }))
    ).resolves.toMatchSnapshot(fixture);
  });
}
