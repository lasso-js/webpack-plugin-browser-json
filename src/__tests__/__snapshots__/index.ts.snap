// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direct-require: direct-require 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

console.log(\\"index\\");
__webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2)

/***/ }),
/* 2 */
/***/ (function(module, exports) {

console.log(\\"other\\");


/***/ })
"
`;

exports[`folder-browser-json-index-file: folder-browser-json-index-file 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);console.log(\\"index\\");


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2)

/***/ }),
/* 2 */
/***/ (function(module, exports) {

console.log(\\"other\\");


/***/ })
"
`;

exports[`nested-directory: nested-directory 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);
console.log(\\"index\\");


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2);console.log(\\"file\\");


/***/ }),
/* 2 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(3)

/***/ }),
/* 3 */
/***/ (function(module, exports) {

console.log(\\"other\\");


/***/ })
"
`;

exports[`remap-issuer-relative: remap-issuer-relative 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

console.log(\\"index\\");
__webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

console.log(\\"remaped\\");


/***/ })
"
`;

exports[`remap-resource-relative: remap-resource-relative 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

console.log(\\"index\\");
__webpack_require__(1);


/***/ }),
/* 1 */
/***/ (function(module, exports) {

console.log(\\"remaped\\");


/***/ })
"
`;

exports[`remap-with-flag-off: remap-with-flag-off 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);console.log(\\"index\\");


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2)

/***/ }),
/* 2 */
/***/ (function(module, exports) {

console.log(\\"other\\");


/***/ })
"
`;

exports[`remap-with-flag-on: remap-with-flag-on 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);console.log(\\"index\\");


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2)

/***/ }),
/* 2 */
/***/ (function(module, exports) {

console.log(\\"remaped\\");


/***/ })
"
`;

exports[`remap-within-browser-json: remap-within-browser-json 1`] = `
"/* 0 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(1);console.log(\\"index\\");


/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(2)

/***/ }),
/* 2 */
/***/ (function(module, exports) {

console.log(\\"remaped\\");


/***/ })
"
`;
